components/ca-crud.html

```diff
@@ -687,12 +687,24 @@
 
             // extracts action information to be used by the UI such as button labels
             actions.forEach(item => {
-
-                // store action using its name as the key, this allows retrive
+                // store action using its name as the key, this allows retrieve
                 // action based on button label clicked from dialog
                 labelToAction[item.title] = item;
             });
 
+            // get a list of navigations from hateoas links with URN nav:name
+            const navs = schemaActions.getNavsFromData(data);
+
+            // extracts navigation information to be used by the UI such as button labels
+            const labelToNav = {};
+
+            // extracts navigation information to be used by the UI such as button labels
+            navs.forEach(item => {
+                // store navigation using its name as the key, this allows retrieve
+                // navigation based on button label clicked from dialog
+                labelToNav[item.title] = item;
+            });
```

Comment:
True. But I'm following the same code pattern as for the actions which is directly above. I find the reduce function harder to understand if I'm honest...